urllib3/poolmanager.py

```diff
@@ -312,8 +312,11 @@ def urlopen(self, method, url, redirect=True, **kw):
 
         kw['assert_same_host'] = False
         kw['redirect'] = False
+
         if 'headers' not in kw:
-            kw['headers'] = self.headers
+            kw['headers'] = headers = self.headers
```

Comment:
since you set `kw['headers']` below, do you need to set it here?

Also, you should probably copy if using `self.headers` so you don't modify that dict.

Reply:
Good catch, I've implemented this.